src/lib.rs

```diff
@@ -572,24 +566,44 @@ fn derive_where_internal(attr: TokenStream, item: TokenStream) -> Result<TokenSt
     } = syn::parse2(item)?;
 
     // Every trait needs a separate implementation.
-    for trait_ in derive_where.traits() {
+    for trait_ in derive_where.traits {
         let body = trait_.generate_body(&ident, &data)?;
-        let trait_ = trait_.type_();
+        let trait_ = trait_.path();
 
         // Build necessary generics to construct the implementation item.
-        let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
-        // TODO: don't convert to `TokenStream`, but actually properly insert bounds.
-        let mut where_clause = where_clause.map(|where_clause| where_clause.into_token_stream());
+        let (impl_generics, type_generics, mut where_clause) = generics.split_for_impl();
+
+        let default_where = &WhereClause {
+            where_token: Where::default(),
+            predicates: Punctuated::default(),
+        };
+
+        let mut where_clause = where_clause.get_or_insert(default_where).clone();
```

Comment:
```suggestion
        let mut where_clause = where_clause
            .cloned()
            .get_or_insert(WhereClause {
                where_token: Where::default(),
                predicates: Punctuated::default(),
            });
```
Use `cloned` first instead of after. Feel free to use a separate variable to instantiate the `default_where`.
This whole block btw should be behind a `if !derive_where.generics.is_empty()`, because otherwise we are adding an empty `where block`.

Reply:
That would also be taken care of in: https://docs.rs/syn/1.0.81/src/syn/generics.rs.html#1297-1304

Reply:
Yeah, I noticed that in my derive tests. But I don't think we should run unnecessary code if we don't plan to use it.